session/session.go

```diff
@@ -312,32 +313,32 @@ func sanitizeBalInfo(balInfo perun.BalInfo) {
 	}
 }
 
-// retrieveParts retrieves the peers from corresponding to the aliases from the contacts provider.
+// retrieveParts retrieves the peer IDs corresponding to the aliases from the ID provider.
 // The order of entries for parts list will be same as that of aliases. i.e aliases[i] = parts[i].Alias.
-func retrieveParts(aliases []string, contacts perun.ContactsReader) ([]perun.Peer, error) {
-	knownParts := make(map[string]perun.Peer, len(aliases))
-	parts := make([]perun.Peer, len(aliases))
+func retrieveParts(aliases []string, idProvider perun.IDReader) ([]perun.PeerID, error) {
```

Comment:
```suggestion
func retrievePartIDs(aliases []string, idProvider perun.IDReader) ([]perun.PeerID, error) {
```

As the function returns `PeerIDs` instead of `Peers`, updating the function name.